lib/matplotlib/pyplot.py

```diff
@@ -1805,7 +1903,13 @@ def ylim(*args, **kwargs):
     return ret
 
 
-def xticks(ticks=None, labels=None, *, minor=False, **kwargs):
+def xticks(
+    ticks: Sequence[float] | None = None,
```

Comment:
Ultimately, this comes down to `axis.Axis.set_ticks` being documented as `list of floats`... I apparently translated that to `Iterable[float]` there(/in Axes), and for some reason slightly narrowed back to `Sequence[float]` for pyplot. The MS type stubs just typed it as `ArrayLike`, which is broader.

Units are a particular case for typing that I haven't gone through and found all of the places where they would apply/expand types. This was partly because they were not being noisy in mypy and partly because I see that as an easier to review change as a perturbation rather than included in full here. (units.py and associated files like dates and category are not actually type hinted yet at all, they don't have any warnings raised and so are separable in that sense, though of course do interweave in many places in Axes/Axis)

So in short, even if it is technically not 100% correct, I think this is a change better made as a follow up. It is consistent with docstrings, so if it should change, the change is likely more than _only_ type hints. (I've been trying to keep this PR hemmed in by that where possible, though a handful of places where mypy warns were addressed)